gcc error when building on CentOS 7

Howdy! Started getting this error when installing the `compass` gem as of today on 1.16.+
Running this on CentOS 7 (wish I didn't and wish I didn't even need compass, but don't get me started... Work project).

```
gcc -I"/opt/rh/rh-ruby26/root/usr/local/share/gems/gems/ffi-1.16.2/ext/ffi_c/libffi-x86_64-linux"/include -I. -I/opt/rh/rh-ruby26/root/usr/include -I/opt/rh/rh-ruby26/root/usr/include/ruby/backward -I/opt/rh/rh-ruby26/root/usr/include -I. -DRUBY_EXTCONF_H=\"extconf.h\"    -fPIC -O2 -g -pipe -Wall -Wp%!(PACKER_COMMA)-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -mtune=generic -fPIC -m64 -o FunctionInfo.o -c FunctionInfo.c
FunctionInfo.c: In function ‘fntype_mark’:
FunctionInfo.c:100:9: error: ‘for’ loop initial declarations are only allowed in C99 mode
         for (size_t index = 0; index < fnInfo->callbackCount; index++) {
         ^
FunctionInfo.c:100:9: note: use option -std=c99 or -std=gnu99 to compile your code
FunctionInfo.c: In function ‘fntype_compact’:
FunctionInfo.c:114:9: error: ‘for’ loop initial declarations are only allowed in C99 mode
         for (size_t index = 0; index < fnInfo->callbackCount; index++) {
         ^
```

--- a/ext/ffi_c/AbstractMemory.c
+++ b/ext/ffi_c/AbstractMemory.c
@@ -350,6 +350,7 @@ memory_get(VALUE self, VALUE type_name, VALUE offset)
     AbstractMemory* ptr;
     VALUE nType;
     Type *type;
+    MemoryOp *op;
 
     nType = rbffi_Type_Lookup(type_name);
     if(NIL_P(nType)) goto undefined_type;
@@ -357,7 +358,7 @@ memory_get(VALUE self, VALUE type_name, VALUE offset)
     TypedData_Get_Struct(self, AbstractMemory, &rbffi_abstract_memory_data_type, ptr);
     TypedData_Get_Struct(nType, Type, &rbffi_type_data_type, type);
 
-    MemoryOp *op = get_memory_op(type);
+    op = get_memory_op(type);
     if(op == NULL) goto undefined_type;
 
     return op->get(ptr, NUM2LONG(offset));
@@ -382,6 +383,7 @@ memory_put(VALUE self, VALUE type_name, VALUE offset, VALUE value)
     AbstractMemory* ptr;
     VALUE nType;
     Type *type;
+    MemoryOp *op;
 
     nType = rbffi_Type_Lookup(type_name);
     if(NIL_P(nType)) goto undefined_type;
@@ -389,7 +391,7 @@ memory_put(VALUE self, VALUE type_name, VALUE offset, VALUE value)
     TypedData_Get_Struct(self, AbstractMemory, &rbffi_abstract_memory_data_type, ptr);
     TypedData_Get_Struct(nType, Type, &rbffi_type_data_type, type);
 
-    MemoryOp *op = get_memory_op(type);
+    op = get_memory_op(type);
     if(op == NULL) goto undefined_type;
 
     op->put(ptr, NUM2LONG(offset), value);

--- a/ext/ffi_c/Call.c
+++ b/ext/ffi_c/Call.c
@@ -427,8 +427,10 @@ getPointer(VALUE value, int type)
     } else if (type == T_DATA && rb_obj_is_kind_of(value, rbffi_StructClass)) {
 
         Struct* s;
+        AbstractMemory* memory;
+
         TypedData_Get_Struct(value, Struct, &rbffi_struct_data_type, s);
-        AbstractMemory* memory = s->pointer;
+        memory = s->pointer;
         return memory != NULL ? memory->address : NULL;
 
     } else if (type == T_STRING) {
@@ -464,7 +466,9 @@ rbffi_GetInvoker(FunctionType *fnInfo)
 static void*
 callback_param(VALUE proc, VALUE cbInfo)
 {
-    VALUE callback ;
+    VALUE callback;
+    AbstractMemory *mem;
+
     if (unlikely(proc == Qnil)) {
         return NULL ;
     }
@@ -479,7 +483,6 @@ callback_param(VALUE proc, VALUE cbInfo)
     callback = rbffi_Function_ForProc(cbInfo, proc);
     RB_GC_GUARD(callback);
 
-    AbstractMemory *mem;
     TypedData_Get_Struct(callback, AbstractMemory, &rbffi_abstract_memory_data_type, mem);
     return mem->address;
 }

--- a/ext/ffi_c/Function.c
+++ b/ext/ffi_c/Function.c
@@ -953,10 +953,10 @@ invoke_callback(VALUE data)
 
             } else if (rb_obj_is_kind_of(rbReturnValue, rb_cProc) || rb_respond_to(rbReturnValue, id_call)) {
                 VALUE function;
+                AbstractMemory* memory;
 
                 function = rbffi_Function_ForProc(rbReturnType, rbReturnValue);
 
-                AbstractMemory* memory;
                 TypedData_Get_Struct(function, AbstractMemory, &rbffi_abstract_memory_data_type, memory);
 
                 *((void **) retval) = memory->address;
@@ -968,8 +968,10 @@ invoke_callback(VALUE data)
         case NATIVE_STRUCT:
             if (TYPE(rbReturnValue) == T_DATA && rb_obj_is_kind_of(rbReturnValue, rbffi_StructClass)) {
                 Struct* s;
+                AbstractMemory* memory;
+
                 TypedData_Get_Struct(rbReturnValue, Struct, &rbffi_struct_data_type, s);
-                AbstractMemory* memory = s->pointer;
+                memory = s->pointer;
 
                 if (memory->address != NULL) {
                     memcpy(retval, memory->address, returnType->ffiType->size);

--- a/ext/ffi_c/FunctionInfo.c
+++ b/ext/ffi_c/FunctionInfo.c
@@ -97,7 +97,8 @@ fntype_mark(void *data)
     rb_gc_mark_movable(fnInfo->rbParameterTypes);
     rb_gc_mark_movable(fnInfo->rbEnums);
     if (fnInfo->callbackCount > 0 && fnInfo->callbackParameters != NULL) {
-        for (size_t index = 0; index < fnInfo->callbackCount; index++) {
+        size_t index;
+        for (index = 0; index < fnInfo->callbackCount; index++) {
             rb_gc_mark_movable(fnInfo->callbackParameters[index]);
         }
     }
@@ -111,7 +112,8 @@ fntype_compact(void *data)
     ffi_gc_location(fnInfo->rbParameterTypes);
     ffi_gc_location(fnInfo->rbEnums);
     if (fnInfo->callbackCount > 0 && fnInfo->callbackParameters != NULL) {
-        for (size_t index = 0; index < fnInfo->callbackCount; index++) {
+        size_t index;
+        for (index = 0; index < fnInfo->callbackCount; index++) {
             ffi_gc_location(fnInfo->callbackParameters[index]);
         }
     }

--- a/ext/ffi_c/Struct.c
+++ b/ext/ffi_c/Struct.c
@@ -189,9 +189,11 @@ struct_initialize_copy(VALUE self, VALUE other)
     }
 
     if (src->layout->referenceFieldCount > 0) {
+        size_t index;
+
         dst->rbReferences = ALLOC_N(VALUE, dst->layout->referenceFieldCount);
         memcpy(dst->rbReferences, src->rbReferences, dst->layout->referenceFieldCount * sizeof(VALUE));
-        for (size_t index = 0; index < dst->layout->referenceFieldCount; index++) {
+        for ( index = 0; index < dst->layout->referenceFieldCount; index++) {
             RB_OBJ_WRITTEN(self, Qundef, &dst->rbReferences[index]);
         }
     }
@@ -268,7 +270,8 @@ struct_mark(void *data)
     rb_gc_mark_movable(s->rbPointer);
     rb_gc_mark_movable(s->rbLayout);
     if (s->rbReferences != NULL) {
-        for (size_t index = 0; index < s->layout->referenceFieldCount; index++) {
+        size_t index;
+        for (index = 0; index < s->layout->referenceFieldCount; index++) {
             rb_gc_mark_movable(s->rbReferences[index]);
         }
     }
@@ -281,7 +284,8 @@ struct_compact(void *data)
     ffi_gc_location(s->rbPointer);
     ffi_gc_location(s->rbLayout);
     if (s->rbReferences != NULL) {
-        for (size_t index = 0; index < s->layout->referenceFieldCount; index++) {
+        size_t index;
+        for (index = 0; index < s->layout->referenceFieldCount; index++) {
             ffi_gc_location(s->rbReferences[index]);
         }
     }

